Node.js client tests failures on windows

Node.js client tests have some failures when running on Windows. Most of test failures seem as intermittent failure (i.e. based on some race condition). We have seen following test failed so far but there can be another intermittent failures (I will add them to this issue when they fail):

### HazelcastClient getDistributedObjects does not return removed object
```
AssertionError: expected [ 'map3', 'map2', 'map1' ] to have the same members as [ 'map2', 'map3' ]
+ expected - actual
10:26:18 
 [
+  "map2"
   "map3"
-  "map2"
-  "map1"
 ]
at test\HazelcastClientTest.js:139:47
at tryCatcher (node_modules\bluebird\js\release\util.js:16:23)
...
```

### NearCachedMap invalidate on change=true containsKey true(in near cache)
```
AssertionError: expected 0 to equal 1
+ expected - actual
-0
+1
at expectStats (test\map\NearCachedMapTest.js:81:43)
at test\map\NearCachedMapTest.js:152:21
...
```

### NearCachedMap invalidate on change=true containsKey false(in near cache)
```
AssertionError: expected 0 to equal 1
+ expected - actual
-0
+1
at expectStats (test\map\NearCachedMapTest.js:81:43)
at test\map\NearCachedMapTest.js:161:21
...

```
### NearCachedMap invalidate on change=true second get should hit
```
AssertionError: expected 2 to equal 1
+ expected - actual
-2
+1
at test\map\NearCachedMapTest.js:92:48
...
```

### Listeners on reconnect restart member, listener still receives map.put event [smart=true]
Error: Timeout of 7000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.

### MigratedData killing a server migrates data to the other node, migrated data has new uuid, near cache discards data with old uuid
```
AssertionError: expected 0 to equal 1
+ expected - actual
-0
+1
at test\nearcache\MigratedDataTest.js:104:39
...
```

### HOW TO REPRODUCE IT
Following jobs are used for testing with particular Node.js version (they are based on similar jobs for running testsuite on linux, but they are executed in through Windows command line):
https://hazelcast-l337.ci.cloudbees.com/job/NodeJS-10-windows
https://hazelcast-l337.ci.cloudbees.com/job/NodeJS-8-windows
https://hazelcast-l337.ci.cloudbees.com/job/NodeJS-6-windows
https://hazelcast-l337.ci.cloudbees.com/job/NodeJS-4-windows

I prepared following jenkins job which can be used for running Node.js tests on windows from your github branch - https://hazelcast-l337.ci.cloudbees.com/job/NodeJS-windows-playground/. See its description for details about what you need to configure.

--- a/test/ListenersOnReconnectTest.js
+++ b/test/ListenersOnReconnectTest.js
@@ -27,6 +27,7 @@ describe('Listeners on reconnect', function () {
     var client;
     var members = [];
     var cluster;
+    var map;
 
     beforeEach(function () {
         return Controller.createCluster(null, null).then(function (cl) {
@@ -35,7 +36,7 @@ describe('Listeners on reconnect', function () {
     });
 
     afterEach(function () {
-        this.timeout(30000);
+        map.destroy();
         client.shutdown();
         return Controller.shutdownCluster(cluster.id);
     });
@@ -43,7 +44,6 @@ describe('Listeners on reconnect', function () {
     [true, false].forEach(function (isSmart) {
 
         function closeTwoMembersOfThreeAndTestListener(done, membersToClose, turnoffMember) {
-            var map;
             Controller.startMember(cluster.id).then(function (m) {
                 members[0] = m;
                 return Controller.startMember(cluster.id);
@@ -116,8 +116,6 @@ describe('Listeners on reconnect', function () {
         });
 
         it('restart member, listener still receives map.put event [smart=' + isSmart + ']', function (done) {
-            this.timeout(7000);
-            var map;
             var member;
             Controller.startMember(cluster.id).then(function (m) {
                 member = m;

